logger/test_logger.py

```diff
@@ -0,0 +1,200 @@
+
+        assert actual == expected
+
+    def test_message_and_metadata_without_frame(self) -> None:
+        """
+        Test with frame is None
+        """
+        frame = None
+        message = "Test message"
+        actual = logger.Logger.message_and_metadata(message, frame)
+
+        expected = "Test message"
+
+        assert actual == expected
+
+
+# Fixtures are used to setup and teardown resources for tests
+# pylint: disable=redefined-outer-name
+class TestLogger:
+    """
+    Test if logger logs the correct messages to file and stdout
+    """
+
+    def test_log_to_file(
+        self,
+        main_logger_instance_and_log_file_path: "tuple[logger.Logger | None, pathlib.Path | None]",
+        logger_instance_to_file_enabled: logger.Logger,
+    ) -> None:
+        """
+        Test if messages are logged to file
+        All levels are done in one test since they will all be logged to the same file
+        """
+        main_logger_instance, log_file_path = main_logger_instance_and_log_file_path
+
+        main_message = "main message"
+        main_logger_instance.debug(main_message, None)
+        main_logger_instance.info(main_message, None)
+        main_logger_instance.warning(main_message, None)
+        main_logger_instance.error(main_message, None)
+        main_logger_instance.critical(main_message, None)
+
+        test_message = "test message"
+        logger_instance_to_file_enabled.debug(test_message, None)
+        logger_instance_to_file_enabled.info(test_message, None)
+        logger_instance_to_file_enabled.warning(test_message, None)
+        logger_instance_to_file_enabled.error(test_message, None)
+        logger_instance_to_file_enabled.critical(test_message, None)
+
+        main_log_file_path = pathlib.Path(log_file_path, "main.log")
+        test_log_file_path = pathlib.Path(log_file_path, "test_logger_to_file_enabled.log")
+
+        with open(main_log_file_path, "r", encoding="utf8") as log_file:
+            actual_main = log_file.read()
+
+        with open(test_log_file_path, "r", encoding="utf8") as log_file:
+            actual_test = log_file.read()
+
+        for level in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]:
+            expected = f"[{level}] {main_message}\n"
+            assert expected in actual_main  # don't know timestamps, so check existance of message
+
+        for level in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]:
+            expected = f"[{level}] {test_message}\n"
+            assert expected in actual_test  # don't know timestamps, so check existance of message
+
+    def test_debug_log_debug_to_stdout(
+        self, caplog: pytest.LogCaptureFixture, logger_instance_to_file_disabled: logger.Logger
+    ) -> None:
+        """
+        Test if debug level message is logged to stdout
+        """
+        test_message = "test message"
+
+        logger_instance_to_file_disabled.debug(test_message, None)
```

Comment:
Same for all of these, would be nice to log twice, one with frame and one without